TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/ArmSystem.java

```diff
@@ -298,150 +196,56 @@ private void toggleGripper() {
-        // fine, just don't run this code. I'm keeping it here because we might want it later,
-        // although calling it in init() or start() will take more time in the round which might
-        // cause problems.
-        // TODO: Make this use init_loop so it doesn't timeout.
-        slider.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
-        slider.setDirection(Direction.motorDirection(direction));
-        slider.setPower(0.1);
-
-        // going up
-        for( int i = 0; i < 1111; i++) {
-            // go up
-            slider.setDirection(Direction.motorDirection(Direction.UP));
-            slider.setPower(0.2);
-            if(limitSwitch.getState()) {
-                break;
-            }
-            try {
-                Thread.sleep(2);
-            } catch (Exception e) {
-
-            }
-
-        }
-        slider.setPower(0);
-        //going down
-        for( int i = 0; i < 1111; i++) {
-            // go up
-            slider.setDirection(Direction.motorDirection(Direction.DOWN));
-            slider.setPower(0.2);
-            if(!limitSwitch.getState()) {
-                break;
-            }
-            try {
-                Thread.sleep(2);
-            } catch (Exception e) {
-
-            }
-        }
-        calibrated = true;
-        this.calibrationDistance = slider.getCurrentPosition();
-        slider.setPower(0);
-        /*
-        // At rest, pre-int, this will be false, i.e. the limit switch IS pressed.
-        if (limitSwitch.getState()) {
-            // Limit switch has been released, cord is under tension.
-            // With cord under tension, lower until switch is depressed.
-            direction = Direction.DOWN;
-        }
-
-        Log.d(TAG,"Calibrating");
-        Log.d(TAG, "Direction: " + direction);
-        Log.d(TAG, "Switch State: " + limitSwitch.getState());
-        // If we're going down and we hit the switch, then we're done
-        if (direction == Direction.DOWN && !limitSwitch.getState()) {
-            slider.setPower(0);
-            calibrated = true;
-            calibrationDistance = slider.getCurrentPosition();
-            slider.setDirection(Direction.motorDirection(Direction.UP));
-            setSliderHeight(0);
-        }
-
-         */
+    private void placeStone () {
+        openGripper();
+        setSliderHeight(getSliderPos() + 1);
+        movePresetPosition(Position.POSITION_HOME);
+        moveHome();
     }
 
-    private void stop() {
-        slider.setPower(0);
+    private void movePresetPosition (Position pos){
+        double[] posArray = pos.getPos();
+        pivot.setPosition(posArray[0]);
+        elbow.setPosition(posArray[1]);
+        wrist.setPosition(posArray[2]);
     }
 
     // Pos should be the # of blocks high it should be
-    private void setSliderHeight(int pos) {
+    public void setSliderHeight ( int pos){
         targetHeight = pos;
```

Comment:
Why not set `targetHeight` to `calculateHeight(pos)`? Then we wouldn't need to recalculate in `updateHeight()`

Reply:
So you can easily add or subtract 1 from it when going up / down

Reply:
Mm. I see that now. How about creating `targetPosition` that is set when `setSliderHeight` is called, and can then be reused every cycle when raise is being called, rather than recalculating each time?